test/benchmark/common/host_guest_bench_runner.sh

```diff
@@ -59,8 +59,21 @@ run_benchmark() {
     # Sleep for a short time to ensure the guest is fully ready
     sleep 1
 
+    # Pin qemu vCPUs to CPUs
+
+    if [[ "$guest_cmd" == *"linux_monitor"* ]]; then
+        bash ${BENCHMARK_DIR}/../../tools/bind_core.sh /run/linux_monitor.socket
+    else
+        bash ${BENCHMARK_DIR}/../../tools/bind_core.sh /run/asterinas_monitor.socket
+    fi
+
+    # Sleep for a short time to ensure taskset is ready
+    sleep 1
+
     # Run the host command and save the output to the specified file.
-    bash "${BENCHMARK_PATH}/host.sh" 2>&1 | tee "${output_file}"  
+    # FIXME: If QEMU starts with more than one vCPUs, 
+    # the CPUs of taskset should be changed accordingly.   
+    taskset -c 1 bash "${BENCHMARK_PATH}/host.sh" 2>&1 | tee "${output_file}"  
```

Comment:
```
    taskset -c 1 bash "${BENCHMARK_PATH}/host.sh" 2>&1 | tee "${output_file}"  
```

So `host.sh` is pin to CPU 1. Is CPU 1 being used by the guest side?

And why do you want to pin the CPUs for `host.sh` in the first place?

```
# FIXME: If QEMU starts with more than one vCPUs, 
```

And since the scripts runs on the host, why does it has anything to do with QEMU?

Reply:
> So host.sh is pin to CPU 1. Is CPU 1 being used by the guest side?

The guest VM is launched with SMP=1 for all benchmarks, meaning the guest vCPU only uses CPU 0（with our `bind_core` script）. The QEMU main process isn't pinned currently, so CPU 1 should be available for the host client.

> And why do you want to pin the CPUs for host.sh in the first place?

I want both the server in the guest VM and the client on the host to be pinned to specific CPUs. This ensures benchmark performance isn't impacted by the host OS scheduling. Since the host client is initiated by `host.sh`, pinning `host.sh` should suffice.

> And since the scripts runs on the host, why does it has anything to do with QEMU?

This is because each QEMU vCPU is pinned to a specific CPU. With SMP=1, the QEMU vCPU only uses CPU 0, allowing the host process to use CPU 1. If SMP is set to a higher value, like 4, vCPUs will be bound to CPUs 0-3, so the host process would need to use CPU 4 to avoid conflicts. I have updated the comments to explain this rationale.